src/lib/metrics.ts

```diff
@@ -360,57 +370,89 @@ export default class MetricsMonitor {
+            featureToggleUpdateTotal
+                .labels(featureName, project, 'n/a', 'n/a')
+                .inc();
         });
         eventStore.on(FEATURE_UPDATED, ({ featureName, project }) => {
             featureToggleUpdateTotal
-                .labels(featureName, project, 'default')
+                .labels(featureName, project, 'default', 'n/a')
                 .inc();
         });
         eventStore.on(
             FEATURE_STRATEGY_ADD,
-            ({ featureName, project, environment }) => {
+            async ({ featureName, project, environment }) => {
+                const environmentType = await this.resolveEnvironmentType(
+                    environment,
+                    cachedEnvironments,
+                );
                 featureToggleUpdateTotal
-                    .labels(featureName, project, environment)
+                    .labels(featureName, project, environment, environmentType)
                     .inc();
             },
         );
         eventStore.on(
             FEATURE_STRATEGY_REMOVE,
-            ({ featureName, project, environment }) => {
+            async ({ featureName, project, environment }) => {
+                const environmentType = await this.resolveEnvironmentType(
+                    environment,
+                    cachedEnvironments,
+                );
                 featureToggleUpdateTotal
-                    .labels(featureName, project, environment)
+                    .labels(featureName, project, environment, environmentType)
                     .inc();
             },
         );
         eventStore.on(
             FEATURE_STRATEGY_UPDATE,
-            ({ featureName, project, environment }) => {
+            async ({ featureName, project, environment }) => {
+                const environmentType = await this.resolveEnvironmentType(
+                    environment,
+                    cachedEnvironments,
+                );
                 featureToggleUpdateTotal
-                    .labels(featureName, project, environment)
+                    .labels(featureName, project, environment, environmentType)
                     .inc();
             },
         );
         eventStore.on(
             FEATURE_ENVIRONMENT_DISABLED,
-            ({ featureName, project, environment }) => {
+            async ({ featureName, project, environment }) => {
+                const environmentType = await this.resolveEnvironmentType(
+                    environment,
+                    cachedEnvironments,
+                );
                 featureToggleUpdateTotal
-                    .labels(featureName, project, environment)
+                    .labels(featureName, project, environment, environmentType)
                     .inc();
             },
         );
         eventStore.on(
             FEATURE_ENVIRONMENT_ENABLED,
             ({ featureName, project, environment }) => {
```

Comment:
should be async for consistency with previous listeners

Reply:
fixed